addons/sale_loyalty/models/sale_order.py

```diff
@@ -74,9 +74,9 @@ def action_confirm(self):
         self._send_reward_coupon_mail()
         return super(SaleOrder, self).action_confirm()
 
-    def action_cancel(self):
+    def _action_cancel(self):
         previously_confirmed = self.filtered(lambda s: s.state in ('sale', 'done'))
-        res = super(SaleOrder, self).action_cancel()
+        res = super()._action_cancel()
```

Comment:
@Williambraecky FYI, this should probably be fixed in 15.3 as well (Cf #81401, it seems you lost the change in your remastering of coupon ;))

Reply:
Here: https://github.com/odoo/odoo/pull/90861 :)